### PR TITLE
Extend websocket next documentation to explain when and when not to subscribe to a Uni or Multi

### LINKED ISSUE #42566: be explicit/use examples in websocket.next docs when it comes to subscribe call for action
### Description

Trying to use websockets.next primiarly the client but i see similar challenges with server side.

reading https://quarkus.io/guides/websockets-next-reference there are a mix of places where it says things like:

"The method must subscribe to the `Multi` to receive these items (or return a Multi)."

and 

"Sends a text message to the client. It returns a `Uni`. It's non-blocking, but you must subscribe to it."

..and no example so is this saying that i.e. it must be `.sendText(msg).subscribe()` or must it be `.sendText(msg).subscribe().with(()->{});` ? we should provide examples in these places to reduce ambiguity/errors.

### Implementation ideas

I would make a PR for these but I'm not actually sure what the required minimal samples are ;)

### PR DESCRIPTION
**Pull request by cescoffier**
Fix https://github.com/quarkusio/quarkus/issues/42566

### CODE REVIEW

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 1]
```diff
@@ -267,7 +267,29 @@ However, it may also accept the following parameters:
 The message object represents the data sent and can be accessed as either raw content (`String`, `JsonObject`, `JsonArray`, `Buffer` or `byte[]`) or deserialized high-level objects, which is the recommended approach.
 
 When receiving a `Multi`, the method is invoked once per connection, and the provided `Multi` receives the items transmitted by this connection.
-The method must subscribe to the `Multi` to receive these items (or return a Multi).
+If the method returns a `Multi` (derived on the received one), Quarkus will automatically subscribe to it and write the emitted items until completion, failure, or cancellation.
```
Comment (mkouba) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:270]: What does _"(derived on the received one)"_ mean?

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 2]
```diff
@@ -1078,7 +1119,17 @@ String param = connection.pathParam("foo");
 The `WebSocketClientConnection` provides both a blocking and a non-blocking method variants to send messages:
 
 - `sendTextAndAwait(String message)`: Sends a text message to the client and waits for the message to be sent. It's blocking and should only be called from an executor thread.
-- `sendText(String message)`: Sends a text message to the client. It returns a `Uni`. It's non-blocking, but you must subscribe to it.
+- `sendText(String message)`: Sends a text message to the client. It returns a `Uni`. It's non-blocking.Make sure you or Quarkus subscribes to the returned `Uni` to send the message.
```
Comment (mkouba) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:1122]:
```suggestion
- `sendText(String message)`: Sends a text message to the client. It returns a `Uni`. It's non-blocking. Make sure you or Quarkus subscribes to the returned `Uni` to send the message.
```

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 3]
```diff
@@ -1203,6 +1254,39 @@ quarkus.log.category."io.quarkus.websockets.next.traffic".level=DEBUG <3>
 <2> Set the number of characters of a text message payload which will be logged.
 <3> Enable `DEBUG` level is for the logger `io.quarkus.websockets.next.traffic`.
 
+[[subscribe-or-not-subscribe]]
+== When to subscribe to a `Uni` or `Multi`
+
+`Uni` and `Multi` are lazy types, which means that they do not start processing until they are subscribed to.
+
+When you call a method returning a `Multi` or `Uni`, or receive a `Multi` or `Uni` as a parameter, whether you should subscribe to it depends on the context:
```
Comment (mkouba) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:1262]: This sentence does not sound right to me. Maybe a native speaker could take a look... @holly-cummins @dmlloyd?

File: docs/src/main/asciidoc/websockets-next-reference.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/websockets-next-reference.adoc]: Applied.

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 4]
```diff
@@ -340,7 +364,8 @@ public Multi<ChatMessage> stream(Multi<ChatMessage> incoming) {
 
 This approach allows bi-directional streaming.
 
-When the method returns `void`, it must subscribe to the incoming `Multi`:
+When the method returns `void`, and so do not return a `Multi`, the code must subscribe to the incoming `Multi`.
```
Comment (mkouba) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:367]:
```suggestion
When the method returns `void`, and so does not return a `Multi`, the code must subscribe to the incoming `Multi`.
``` 
??

**Review by mkouba — COMMENTED** [review]
Looks good.

File: docs/src/main/asciidoc/websockets-next-reference.adoc (hunk 4 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/websockets-next-reference.adoc]: Applied, thanks!

**Review by mkouba — APPROVED** [review]

**Pull request merged**